empress_gui.py

```diff
@@ -4,362 +4,819 @@
+        self.recon_nums_frame.grid_columnconfigure(0, weight=1)
+        self.recon_nums_frame.grid_columnconfigure(1, weight=7)
+        self.recon_nums_frame.grid_propagate(False)
+
+        # To occupy empty space so the grey background in self.master is not shown
+        self.empty_frame = tk.Frame(self.master)
+        self.empty_frame.grid(row=4, column=1, sticky="nsew")
+
+        # To overwrite everything when user loads in new input files 
+        # (always starting from the host tree file)
+        self.host_tree_info = tk.Label(self.input_info_frame)
+        self.parasite_tree_info = tk.Label(self.input_info_frame)
+        self.mapping_info = tk.Label(self.input_info_frame)
+
+        self.dup_label = tk.Label(self.costs_frame)
+        self.dup_entry_box = CustomEntry(self.costs_frame)
+        self.trans_label = tk.Label(self.costs_frame)
+        self.trans_entry_box = tk.Entry(self.costs_frame)
+        self.loss_label = tk.Label(self.costs_frame)
+        self.loss_entry_box = tk.Entry(self.costs_frame)
+        self.dup_input = tk.DoubleVar()
+        self.dup_input.set(1.00)
+        self.trans_input = tk.DoubleVar()
+        self.trans_input.set(1.00)
+        self.loss_input = tk.DoubleVar()
+        self.loss_input.set(1.00)
+
+        self.recon_MPRs_label = tk.Label(self.recon_nums_frame)
+        self.num_MPRs_label = tk.Label(self.recon_nums_frame)
+        self.recon_cospeci_label = tk.Label(self.recon_nums_frame)
+        self.recon_dup_label = tk.Label(self.recon_nums_frame)
+        self.recon_trans_label = tk.Label(self.recon_nums_frame)
+        self.recon_loss_label = tk.Label(self.recon_nums_frame)
+
+        self.num_cluster_input = tk.IntVar()
+        self.num_cluster = None
+
+        self.recon_info_displayed = False
+        self.recon_input = input_reader.ReconInput()
+        App.recon_graph = None
+        App.clusters_list = []
+        App.medians = None
+
+        self.cost_space_window = None
+        self.entire_space_window = None
+        self.set_num_cluster_window = None
+        self.one_MPR_window = None
+        self.view_solution_space_window = None
+        self.view_reconciliations_window = None
+        self.view_pvalue_histogram_window = None
+
+    def reset(self, event):
+        """Reset when user loads in a new input file (can be either of the three options)."""
+        App.recon_graph = None 
+        App.clusters_list = []
+        App.medians = None
+        # Reset self.recon_input so self.view_cost_space_btn can be disabled
+        self.recon_input = input_reader.ReconInput()
+        self.view_cost_space_btn.configure(state=tk.DISABLED)
+        self.view_tanglegram_btn.configure(state=tk.DISABLED)
+
+        if event == "Load host tree file":
+            self.load_files_dropdown['menu'].entryconfigure("Load parasite tree file", state = "normal")
+
+            # Read in host tree file after reseting everything
+            self.recon_input.read_host(self.host_file_path)
```

Comment:
Reading the host tree doesn't sound like part of reset to me. Maybe you want to rename the function.

Reply:
This is in reset() because if the user starts at loading a new parasite tree file again, the old host tree file has to be kept. self.recon_input = input_reader._ReconInput() happened before this.

Reply:
Renamed to refresh_when_new_input_files_loaded